scripts/create_ai_entries.py

```diff
@@ -0,0 +1,233 @@
+                "Terminating program"
+            ),
+            f"{args.pronto}/login/"
+        )
+        sys.exit(1)
+    else:
+        logger.warning("Logged in as %s", username)
+
+    create_entries(s, args)
+
+
+def is_human_complete(sig_response: dict) -> bool:
+    """Check if human curated data is complete"""
+    keys = ["name", "description", "abstract"]
+    return all(sig_response.get(key) for key in keys)
+
+
+def is_llm(sig_response: dict) -> bool:
+    """Check if record is ai-generated:
+        It should have at least one ai-generated element
+    """
+    keys = ["llm_name", "llm_description", "llm_abstract"]
+    return any(sig_response.get(key) for key in keys)
+
+
+def write_error(message, args):
+    if args.cache:
+        with open("failed_signatures.tsv", "a") as fh:
+            fh.write(message)
+    else:
+        sys.stderr.write(message)
+
+
+def create_entries(
+    s: requests.Session,
+    args: argparse.Namespace,
+) -> None:
+    """
+    Attempt to create entries for ai generated signatures.
+
+    :param s: requests session which is logged into pronto
+    :param args: cli args parser
+    """
+    submission_url = f"{args.pronto}/api/entry/"
+
+    for database in args.databases:
+        query_url = (
+            f"{args.pronto}/api/database/{database}/signatures/?integrated=0&commented=0"
+            f"&sort-by=reviewed-proteins&sort-order=desc&page_size={args.batch_size}"
+        )
+        sigs_query = s.get(query_url)
+
+        if sigs_query.status_code != 200:
+            logger.error(
+                "Failed to retrieve unintergated ai-generated signatures for %s:\n%s",
+                database,
+                sigs_query.text,
+            )
+            continue
+
+        sig_total, sig_count = len(sigs_query.json()['results']), 0
+        sig_perc = {f"{i*10}%": round(sig_total * i/10) for i in range(1, 11)}
+
+        for signature in sigs_query.json()['results']:
+            perc = (sig_count / sig_total * 100)
+            if perc in sig_perc.values():
+                logger.info("Parserd %s of signatures", sig_perc[perc])
+
+            sig_acc = signature['accession']
+            sig_type = signature['type']['code']
+            sig_url = f"{args.pronto}/api/signature/{sig_acc}/"
+
+            sig_response = s.get(sig_url)
+
+            if sig_response.status_code != 200:
+                logger.error("Could not retrieve data for signature %s", sig_acc)
+                continue
+
+            if is_human_complete(sig_response.json()):
+                # human curatored data is complete
+                # leave to curator to choose between human- and ai-generated data
+                continue  
+
+            if is_llm(sig_response.json()) is False:
+                continue
+
+            if sig_response.json()['llm_description'] is not None:
+                if len(sig_response.json()['llm_description']) > 30:
```

Comment:
Historically, signatures had a _name_ (short name, indentifier-like, e.g. `CoV_NSP2_C`), a _description_ (e.g. `Coronavirus replicase NSP2, C-terminal`), and an _abstract_.

Unfortunately, these fields were renamed for InterPro entries, so _name_ became _short name_, _description_ became _name_, and _abstract_ became _common annotation block_ and/or _description_.

```suggestion
            if sig_response.json()['llm_name'] is not None:
                if len(sig_response.json()['llm_name']) > 30:
```